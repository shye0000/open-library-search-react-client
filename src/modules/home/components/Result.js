import React from 'react';
import Card from 'antd/lib/card';
import './Result.scss';

const Result = ({result}) => {
	const {
		title,
		author_name,
		first_publish_year,
		cover_i,
		key
	} = result;
	// const id = key.replace('/works/', '');
	const { Meta } = Card;
	return <Card
		hoverable
		style={{ width: 240 }}
		cover={<img alt="cover" src={`https://covers.openlibrary.org/w/id/${cover_i}.jpg`} />}
	>
		<Meta
			title="Europe Street beat"
			description="www.instagram.com"
		/>
	</Card>;
};

export default Result;