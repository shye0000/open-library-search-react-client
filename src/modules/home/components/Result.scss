@import "../../../styles/colors";
.result {
  .ant-card-cover {
    position: relative;
    width: 100%;
    height: 300px;
    background: $background-grey;
    overflow: hidden;
    img.cover {
      position: absolute;
      width: 100%;
      height: auto;
      top: 50%;
      transform: translateY(-50%);
      text-align: center;
    }
  }
  .ant-card-meta-description {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    height: 21px;
    line-height: 21px;
  }
}