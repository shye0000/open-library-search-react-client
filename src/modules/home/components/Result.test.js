import React from 'react';
import Result from './Result';
import { shallow } from 'enzyme';

describe('Result', () => {

	const props = {
		result: {
			title: 'title',
			author_name: ['author_name_1', 'author_name_2'],
			first_publish_year: 2018,
			cover_i: 'COVER_I',
			key: 'key'
		}
	};

	it('render correctly', () => {
		const wrapper = shallow(<Result {...props} />);
		expect(wrapper).toMatchSnapshot();
	});

});