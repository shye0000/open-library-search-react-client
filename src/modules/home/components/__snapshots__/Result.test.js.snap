// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Result
    result={
      Object {
        "author_name": Array [
          "author_name_1",
          "author_name_2",
        ],
        "cover_i": "COVER_I",
        "first_publish_year": 2018,
        "key": "key",
        "title": "title",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Card
        className="result"
        cover={
          <img
            align="middle"
            alt="cover"
            className="cover"
            src="https://covers.openlibrary.org/w/id/COVER_I.jpg"
          />
        }
        hoverable={true}
      >
        <Unknown
          title={2018}
        />
        <Unknown
          description="author_name_1, author_name_2"
          title="title"
        />
      </Card>,
      "replace": false,
      "to": "/book/key",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown
            title={2018}
          />,
          <Unknown
            description="author_name_1, author_name_2"
            title="title"
          />,
        ],
        "className": "result",
        "cover": <img
          align="middle"
          alt="cover"
          className="cover"
          src="https://covers.openlibrary.org/w/id/COVER_I.jpg"
        />,
        "hoverable": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": 2018,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "description": "author_name_1, author_name_2",
            "title": "title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Card
          className="result"
          cover={
            <img
              align="middle"
              alt="cover"
              className="cover"
              src="https://covers.openlibrary.org/w/id/COVER_I.jpg"
            />
          }
          hoverable={true}
        >
          <Unknown
            title={2018}
          />
          <Unknown
            description="author_name_1, author_name_2"
            title="title"
          />
        </Card>,
        "replace": false,
        "to": "/book/key",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Unknown
              title={2018}
            />,
            <Unknown
              description="author_name_1, author_name_2"
              title="title"
            />,
          ],
          "className": "result",
          "cover": <img
            align="middle"
            alt="cover"
            className="cover"
            src="https://covers.openlibrary.org/w/id/COVER_I.jpg"
          />,
          "hoverable": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": 2018,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "description": "author_name_1, author_name_2",
              "title": "title",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
